### PR TITLE
feat(mysql): implement _get_schema_using_query

### LINKED ISSUE #2002: 'MySQLClient' object has no attribute '_get_schema_using_query'
```
In [57]: mysql_con = ibis.mysql.connect(user='user', password='password', host='xx', database='xx') 
In [62]: mysql_con.sql('select * from table1')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-62-657a97131ae3> in <module>
----> 1 mysql_con.sql('select * from table1') 

~/.conda/envs/data_extractor/lib/python3.6/site-packages/ibis/client.py in sql(self, query) 
      139         # there is already a limit in the query, we find and remove it  
      140         limited_query = 'SELECT * FROM ({}) t0 LIMIT 0'.format(query)
--> 141         schema = self._get_schema_using_query(limited_query)
      142         return ops.SQLQueryResult(query, schema, self).to_expr()
      143 
AttributeError: 'MySQLClient' object has no attribute '_get_schema_using_query' 
In [63]: mysql_con.raw_sql('select * from table1', results=True).fetchall()
Out[63]: [(1, 'yh1'), (2, 'yh2'), (3, 'yh3')]
In [64]: ibis.__version__
Out[64]: '1.2.0'
```

### PR DESCRIPTION
**Pull request by cpcloud**
This PR implements `_get_schema_using_query` for the MySQL backend.

I originally was going to add this in #3642, but it became significantly complex
that I broke it out into a separate PR.

Closes #2002.

### CODE REVIEW

File: ibis/backends/mysql/tests/test_client.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import ibis
+import ibis.expr.datatypes as dt
+
+
+@pytest.mark.parametrize(
+    ("mysql_type", "expected_type"),
+    [
+        param(mysql_type, ibis_type, id=mysql_type.lower())
+        for mysql_type, ibis_type in [
```
Comment (kszucs) [inline ibis/backends/mysql/tests/test_client.py:12]: List as a global variable?

File: ibis/backends/mysql/__init__.py [hunk 2]
```diff
@@ -121,6 +124,215 @@ def begin(self):
...
+    "MEDIUM_BLOB": dt.Binary,
+    "LONG_BLOB": dt.Binary,
+    "BLOB": dt.Binary,
+    "VAR_STRING": dt.String,
+    "STRING": dt.String,
+    "GEOMETRY": dt.Geometry,
+}
+
+
+class _FieldFlags:
```
Comment (kszucs) [inline ibis/backends/mysql/__init__.py:282]: Can you move some of this code to `mysql/datatypes.py` by encapsulating the type name and field flags into a `MySQLDataType` or something similar to [ClickhouseDataType](https://github.com/ibis-project/ibis/blob/master/ibis/backends/clickhouse/client.py#L40) (which eventually should be moved to its own file as well)?

Then we could register `dt.dtype(MySQLDataType)` for converting from MySQL to ibis.

Comment (cpcloud) [reply ibis/backends/mysql/__init__.py:282]: I can move this into `mysql/datatypes.py`, but I'd rather avoid doing a big refactoring/restructuring of this code for now. I'm happy to revisit once we see more use of this feature in the wild.

File: ibis/backends/mysql/__init__.py [hunk 3]
```diff
@@ -121,6 +124,26 @@ def begin(self):
...
+        """Infer the schema of `query`."""
+        result = self.con.execute(f"SELECT * FROM ({query}) _ LIMIT 0")
+        cursor = result.cursor
+        fields = [
+            (field.name, _type_from_cursor_info(descr, field))
+            for descr, field in zip(cursor.description, cursor._result.fields)
+        ]
+        return sch.Schema.from_tuples(fields)
+
+    def _get_temp_view_definition(
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/mysql/__init__.py:137]: Ah, it'll be used in the `.sql` PR, let me factor this part out

File: ibis/backends/mysql/tests/test_client.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/mysql/tests/test_client.py]: Done.

File: ibis/backends/mysql/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/mysql/__init__.py]: Fixed.

File: ibis/backends/mysql/datatypes.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    "MEDIUM_BLOB": dt.Binary,
+    "LONG_BLOB": dt.Binary,
+    "BLOB": dt.Binary,
+    "VAR_STRING": dt.String,
+    "STRING": dt.String,
+    "GEOMETRY": dt.Geometry,
+}
+
+
+class _FieldFlags:
```
Comment (kszucs) [inline ibis/backends/mysql/datatypes.py:146]: Just a nit, but could we use python `IntFlag` or [py-flags](https://github.com/pasztorpisti/py-flags) here? We can certainly defer it to a follow-up.

**Review by kszucs — APPROVED** [review]

File: ibis/backends/mysql/datatypes.py (hunk 4 above)
Comment (cpcloud) [reply ibis/backends/mysql/datatypes.py:146]: Ah, forgot about `IntFlag`

Comment (cpcloud) [reply ibis/backends/mysql/datatypes.py:146]: We can remove some of the mapped flags, as they aren't used in practice.

Comment (cpcloud) [reply ibis/backends/mysql/datatypes.py:146]: I will explore this in a follow up, subclassing `IntFlag` requires a good amount of refactoring.

**Pull request merged**